compositor_render/src/renderer/texture.rs

```diff
@@ -93,46 +124,82 @@ impl InnerNodeTexture {
             bind_group: Arc::new(bind_group),
         }
     }
+
+    pub fn rgba_texture(&self) -> &RGBATexture {
+        &self.texture
+    }
+
+    pub fn bind_group(&self) -> &wgpu::BindGroup {
+        &self.bind_group
+    }
+
+    pub fn resolution(&self) -> Resolution {
+        texture_size_to_resolution(&self.texture.size())
+    }
+}
+
+struct InnerNodeTexture {
+    state: Option<NodeTextureState>,
+    is_empty: bool,
```

Comment:
Instead of having `Option` here, use custom Enum with 3 possible values, representing 3 possible states, and implement protected access to it (like getters). `Option` would be suitable, if there was only one possible reason form the lack of access to `NodeTextureState, while in fact there are two.